nixos/modules/installer/cd-dvd/iso-image.nix

```diff
@@ -185,33 +188,25 @@ let
       # So instead we'll list a lot of possibly valid modes :/
       #"3840x2160"
       #"2560x1440"
+      "1920x1200"
       "1920x1080"
       "1366x768"
+      "1280x800"
       "1280x720"
+      "1200x1920"
       "1024x768"
+      "800x1280"
       "800x600"
       "auto"
     ]}
 
-    # Fonts can be loaded?
-    # (This font is assumed to always be provided as a fallback by NixOS)
-    if loadfont (\$root)/EFI/boot/unicode.pf2; then
-      set with_fonts=true
-    fi
-    if [ "\$textmode" != "true" -a "\$with_fonts" == "true" ]; then
-      # Use graphical term, it can be either with background image or a theme.
-      # input is "console", while output is "gfxterm".
-      # This enables "serial" input and output only when possible.
-      # Otherwise the failure mode is to not even enable gfxterm.
-      if test "\$with_serial" == "yes"; then
-        terminal_output gfxterm serial
```

Comment:
is there a reason serial i/o is getting removed?

Reply:
It's already depending on the UEFI console feature to get up to the point where one can use serial [on UEFI]. So to get serial working in grub, you already had to have the UEFI console going to that serial port. So by relying on the intrinsic UEFI console support, it already fulfills the requirement for serial.

Also it's what was the cause of at least one issue.

On legacy boot, GRUB is not involved at all.